lib/union_of.rb

```diff
@@ -0,0 +1,299 @@
+# frozen_string_literal: true
+
+require 'active_record_union'
+
+module UnionOf
```

Comment:
Currently, eager loading is not supported. Should we add support or raise?

Reply:
Here's where we can do an extension: https://github.com/rails/rails/blob/main/activerecord/lib/active_record/associations/preloader/branch.rb#L137

Reply:
Correction: eager loading does not work because there's no way to resolve the inverses of the eager loaded records without loading the associations themselves.

Reply:
Eager loading needs to be solved due to #792.